---
title: "Gene expression"
author: "Rahmat"
date: "2024-11-07"
output: html_document
---

## Gene Expression analysis
```{r}
setwd("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts")
```

# IMPORTING COUNTS
```{r}
SRR1313090 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313090.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313090)
```
```{r}
SRR1313091 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313091.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313091)
```
```{r}
SRR1313092 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313092.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313092)
```
```{r}
SRR1313093 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313093.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313093)
```
```{r}
SRR1313148 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313148.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313148)
```
```{r}
SRR1313149 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313149.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313149)
```
```{r}
SRR1313150 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313150.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313150)
```
```{r}
SRR1313151 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313151.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
```
```{r}
SRR1313204 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313204.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313204)
```
```{r}
SRR1313205 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313205.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR1313206 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313206.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313206)
```
```{r}
SRR1313208 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR1313208.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)
head(SRR1313208)
```
```{r}
SRR13873745 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR13873745.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR13873746 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR13873746.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR13873747 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR13873747.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR13873748 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR13873748.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR22891574 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR22891574.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR22891592 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR22891592.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR22891594 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR22891594.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```
```{r}
SRR22891598 <- read.table("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Counts/SRR22891598.tabular", header= TRUE, row.names = 1, stringsAsFactors = TRUE)

```

# Merging counts data
```{r}
count_data <- cbind(SRR1313090,SRR1313091,SRR1313092,SRR1313093,SRR13873745,SRR13873746,SRR13873747,SRR13873748,SRR1313148,SRR1313149,SRR1313150,SRR1313151,SRR1313204,SRR1313205,SRR1313206,SRR1313208,SRR22891574,SRR22891592,SRR22891594,
SRR22891598)
head(count_data)
```
```{r}
count_data <- count_data[!duplicated(rownames(count_data)), ]
head(count_data)
```
```{r}
library(readxl)
phenodata <- read_excel("C:/Users/rahma/Documents/Breast cancer RNA-Seq/Metadata.xlsx")
phenodata
```
Confirm the column names in counts data tallies with the rownames in the phenodata
```{r}
rownames(phenodata) = phenodata$`SRA Accession number`
phenodata
```
```{r}
rownames(phenodata)
```
```{r}
colnames(count_data)
```
```{r}
colnames(count_data) == rownames(phenodata)
```
```{r}
phenodata$Subtype <- factor(phenodata$Subtype)
class(phenodata$Subtype)
phenodata$Subtype
```
```{r}
phenodata$Subtype = relevel(phenodata$Subtype, ref = "Normal")
phenodata$Subtype
```

# Creating a DESeq dataset
```{r}
library(DESeq2)
library(genefilter)
```
```{r}
dds <- DESeqDataSetFromMatrix(countData = count_data, colData = phenodata, design = ~Subtype)
dds
```
```{r}
smallestGroupSize <- 3
keep <- rowSums(counts(dds) >= 10) >= smallestGroupSize
dds <- dds[keep,]
dds
```
```{r}
levels(dds$Subtype)

```
```{r}
design(dds)
```

# Exploratory data analysis
.
## Data transformation and Variance stabilization
```{r}
vsd <- vst(dds)
vsd
```
```{r}
library(vsn)
meanSdPlot(assay(vsd))
```
```{r}
plotPCA(vsd, intgroup = "Subtype")
```
```{r}
plotPCA(vsd, intgroup= "Location")
```

# Differential gene expression analysis
```{r}
dds1 <- DESeq(dds)
```

```{r}
TNBCvsNorm= results(dds1)
TNBCvsNorm
write.csv(TNBCvsNorm, file= "TNBC.csv")

```
```{r}
TNBCvsNorm= na.omit(TNBCvsNorm)
TNBC= TNBCvsNorm[abs(TNBCvsNorm$log2FoldChange) > 0 & TNBCvsNorm$padj < 0.01,]
TNBC
```

### Collecting the gene IDs from org.Hs.eg.db

```{r}
library(org.Hs.eg.db)
keys= rownames(TNBC)
symbolsTNBC= select(org.Hs.eg.db, keys = keys, keytype = "ENTREZID", columns = c("SYMBOL", "GENENAME"))
head(symbolsTNBC)
```

```{r}
TNBC= cbind(TNBC, symbolsTNBC$SYMBOL, symbolsTNBC$GENENAME)
TNBC = data.frame(TNBC)
head(TNBC)
```
```{r}
orderedTNBC= TNBC[order(TNBC$padj),]
orderedTNBC
```
```{r}
annotation_TNBC <- phenodata[9:16,]
annotation_TNBC
```
```{r}
TNBCidx <- which(TNBCvsNorm$padj <=0.01 & abs(TNBCvsNorm$log2FoldChange)>1)
matTNBC <- assay(vsd)[rownames(vsd) %in% rownames(TNBCvsNorm[TNBCidx,]),][, 9:16]
matTNBC = as.matrix(matTNBC)
head(matTNBC)
```
```{r}
library(pheatmap)
pheatmap(matTNBC[1:20,])
```


```{r}
upregTNBC= orderedTNBC[orderedTNBC$log2FoldChange >0,]
upregTNBC
```
```{r}
downregTNBC <-orderedTNBC[orderedTNBC$log2FoldChange <0,]
downregTNBC

```
```{r}
library(EnhancedVolcano)
EnhancedVolcano(TNBC, lab= TNBC$symbolsTNBC.SYMBOL, x="log2FoldChange", y= 'padj', title= "Normal vs TNBC")
```

```{r}
resultsNames(dds1)
```

```{r}
resNorm_ERPR <- results(dds1, name="Subtype_ER.PR._vs_Normal")
resNorm_ERPR
write.csv(resNorm_ERPR, file = "ERPR.csv")
```
```{r}
resNorm_ERPR = na.omit(resNorm_ERPR)
ERPR = resNorm_ERPR[abs(resNorm_ERPR$log2FoldChange) >= 0 & resNorm_ERPR$padj < 0.01,]
ERPR
```
```{r}
keys= rownames(ERPR)
symbolERPR= select(org.Hs.eg.db, keys = keys, keytype = "ENTREZID", columns = c("SYMBOL", "GENENAME"))
head(symbolERPR)
```

```{r}
ERPR = cbind(ERPR, symbolERPR$SYMBOL, symbolERPR$GENENAME)
ERPR
```
```{r}
upregERPR = ERPR[ERPR$log2FoldChange > 0 & ERPR$padj < 0.01,]
upregERPR= upregERPR[order(upregERPR$log2FoldChange, decreasing = TRUE),]
upregERPR
```

```{r}
downregERPR = ERPR[ERPR$log2FoldChange < 0 & ERPR$padj < 0.01,]
downregERPR= downregERPR[order(downregERPR$log2FoldChange, decreasing = FALSE),]
downregERPR
```
```{r}
EnhancedVolcano(ERPR, lab = ERPR$`symbolERPR$SYMBOL`, x= "log2FoldChange", y= "padj")
```
```{r}
ERPRidx <- which(resNorm_ERPR$padj <=0.01 & abs(resNorm_ERPR$log2FoldChange)>1)
matERPR <- assay(vsd)[rownames(vsd) %in% rownames(resNorm_ERPR[ERPRidx,]),][, c(5:8, 13:16)]
matERPR = as.matrix(matERPR)
head(matERPR)
```
```{r}
pheatmap(matERPR[1:20,])
```

```{r}
resNorm_ER <- results(dds1, name= "Subtype_ER._vs_Normal")
resNorm_ER
write.csv(resNorm_ER, file = "ER.csv")
```
```{r}
resNorm_ER= na.omit(resNorm_ER)
ER = resNorm_ER[abs(resNorm_ER$log2FoldChange) > 0 & resNorm_ER$padj < 0.01 ,]
ER
```
```{r}
keys = rownames(ER)
symbolER = select(org.Hs.eg.db, keys = keys, keytype = "ENTREZID", columns = c("SYMBOL", "GENENAME"))
head(symbolER)
```

```{r}
ER = cbind(ER, symbolER$SYMBOL, symbolER$GENENAME)
ER
```
```{r}
ERidx <- which(resNorm_ER$padj <=0.01 & abs(resNorm_ER$log2FoldChange)>1)
matER <- assay(vsd)[rownames(vsd) %in% rownames(resNorm_ER[ERidx,]),][, c(1:4, 13:16)]
matER = as.matrix(matER)
head(matER)
```
```{r}
pheatmap(matER[1:20,])
```

```{r}
upregER = ER[ER$log2FoldChange > 0,]
upregER = upregER[order(upregER$log2FoldChange, decreasing = TRUE),]
upregER
```
```{r}
downregER = ER[ER$log2FoldChange < 0,]
downregER = downregER[order(downregER$log2FoldChange, decreasing = F),]
downregER
```
```{r}
EnhancedVolcano(ER, lab = ER$`symbolER$SYMBOL`, x= "log2FoldChange", y= "padj")
```

```{r}
resNorm_HER <- results(dds1, name="Subtype_HER2._vs_Normal")
resNorm_HER
write.csv(resNorm_HER, file = "HER2.csv")
```
```{r}
resNorm_HER= na.omit(resNorm_HER)
HER = resNorm_HER[abs(resNorm_HER$log2FoldChange) > 0 & resNorm_HER$padj < 0.01,]
HER
```
```{r}
keys= rownames(HER)
symbolHER = select(org.Hs.eg.db, keys = keys, keytype = "ENTREZID", columns = c("SYMBOL", "GENENAME"))
head(symbolHER)
```
```{r}
HER = cbind(HER, symbolHER$SYMBOL, symbolHER$GENENAME)
HER
```
```{r}
HERidx <- which(resNorm_HER$padj <=0.01 & abs(resNorm_HER$log2FoldChange)>1)
matHER <- assay(vsd)[rownames(vsd) %in% rownames(resNorm_HER[HERidx,]),][, c(1:4, 13:16)]
matHER = as.matrix(matHER)
head(matHER)
```
```{r}
pheatmap(matHER[1:20,])
```

```{r}
upregHER= HER[HER$log2FoldChange > 0,]
upregHER = upregHER[order(upregHER$log2FoldChange, decreasing = TRUE),]
upregHER
```
```{r}
top20upregHER = upregHER[1:20, 7:8]
top20upregHER
```

```{r}
downregHER= HER[HER$log2FoldChange < 0,]
downregHER = downregHER[order(downregHER$log2FoldChange, decreasing = F),]
downregHER
```
```{r}
top20downregHER = downregHER[1:20, 7:8]
top20downregHER
```

```{r}
EnhancedVolcano(HER, lab= HER$`symbolHER$SYMBOL`, x= "log2FoldChange", y= "padj")
```

```{r}
sessionInfo()
```





